File: Leetcode_with_py/104maxDepth.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:
            return 0
        return 1 + max(self.maxDepth(root.left), self.maxDepth(root.right))


class Solution(object):
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:
            return 0
        queue = []
        queue.append(root)
        ans = 0
        while queue:
            ans += 1
            for i in range(len(queue)):
                node = queue.pop(0)  # 0 ≤ªø……Ÿ
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
        return ans

File: Leetcode_with_Cpp/104maxDepth.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        if (!root) return 0;
        return max(maxDepth(root->left), maxDepth(root->right)) + 1;
    }
};

class Solution {
public:
    int maxDepth(TreeNode* root) {
        if (!root) return 0;
        int num = 0;
        queue<TreeNode *> q;
        q.push(root);
        while (!q.empty()){
            int n = q.size();
            for (int i = 0; i < n; ++i) {
                TreeNode *cur = q.front();
                if (cur->left) q.push(cur->left);
                if (cur->right) q.push(cur->right);
                q.pop();
            }
            num++;
        }
        return num;
    }
};